### ISSUE TITLE
E0106 not detected for const in trait or impl block

### ISSUE THREAD

**Issue by ttaomae** [intellij-rust/intellij-rust#1983]
Examples:
```rust
trait T {
    const C: &str;
}
```
```rust
struct S;
impl S {
    const C: &str = "foo;
}
```
In both examples there should be an error on the `&`.
